eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/QueryTypePass.php

```diff
@@ -1,45 +1,74 @@
 <?php
+
 /**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  */
+declare(strict_types=1);
+
 namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
 
+use AppendIterator;
+use ArrayIterator;
+use Iterator;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Reference;
 
 /**
- * Processes services tagged as ezpublish.query_type, and registers them with ezpublish.query_type.registry.
+ * Processes services tagged as ezplatform.query_type, and registers them with ezpublish.query_type.registry.
  */
-class QueryTypePass implements CompilerPassInterface
+final class QueryTypePass implements CompilerPassInterface
 {
-    public const QUERY_TYPE_SERVICE_TAG = 'ezpublish.query_type';
+    public const QUERY_TYPE_SERVICE_TAG = 'ezplatform.query_type';
+    public const DEPRECATED_QUERY_TYPE_SERVICE_TAG = 'ezpublish.query_type';
 
-    public function process(ContainerBuilder $container)
+    public function process(ContainerBuilder $container): void
     {
         if (!$container->hasDefinition('ezpublish.query_type.registry')) {
             return;
         }
 
         $queryTypes = [];
-        $queryTypesClasses = [];
 
-        // tagged query types
-        $taggedServiceIds = $container->findTaggedServiceIds(self::QUERY_TYPE_SERVICE_TAG);
-        foreach ($taggedServiceIds as $taggedServiceId => $tags) {
+        $iterator = $this->getTaggedServiceIdsIterator($container);
+        foreach ($iterator as $taggedServiceId => $tags) {
             $queryTypeDefinition = $container->getDefinition($taggedServiceId);
             $queryTypeClass = $container->getParameterBag()->resolveValue($queryTypeDefinition->getClass());
 
-            $queryTypesClasses[$queryTypeClass] = [];
-
             for ($i = 0, $count = count($tags); $i < $count; ++$i) {
                 $name = isset($tags[$i]['alias']) ? $tags[$i]['alias'] : $queryTypeClass::getName();
                 $queryTypes[$name] = new Reference($taggedServiceId);
-                $queryTypesClasses[$queryTypeClass][$name] = true;
             }
         }
 
         $aggregatorDefinition = $container->getDefinition('ezpublish.query_type.registry');
         $aggregatorDefinition->addMethodCall('addQueryTypes', [$queryTypes]);
     }
+
+    private function getTaggedServiceIdsIterator(ContainerBuilder $container): Iterator
+    {
+        $serviceIdsWithDeprecatedTags = $container->findTaggedServiceIds(
+            self::DEPRECATED_QUERY_TYPE_SERVICE_TAG
+        );
+
+        foreach ($serviceIdsWithDeprecatedTags as $serviceId => $tags) {
+            @trigger_error(
+                sprintf(
+                    'Service tag `%s` is deprecated and will be removed in eZ Platform 4.0. Use `%s` instead.',
```

Comment:
The warning triggered multiple times is not very useful unless you provide `$serviceId` so it's easy to find DI config causing this ;)
E.g.:
```suggestion
                    'Service tag `%s` is deprecated and will be removed in eZ Platform 4.0. Tag `%s` with `%s` instead.',
```
where the second `%s` is `$serviceId`

Reply:
Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3002/commits/2433ef60e37068c928ae630c0dd07019ea79b75b